pkg/notifier/lib.go

```diff
@@ -0,0 +1,46 @@
+package notifier
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/appscode/kubed/pkg/notifier/extpoints"
+)
+
+const (
+	Notify_Via string = "notify_via"
+)
+
+type Notifier struct {
+	configuration map[string]string
+}
+
+func New(conf map[string]string) Notifier {
+	return Notifier{
+		configuration: conf,
+	}
+}
+
+func (n Notifier) SendNotification(notification string) (string, error) {
+	driver, err := n.notificationDriver()
+	if err != nil {
+		return "", err
+	}
+	return driver.Uid(), driver.Notify(notification)
+}
+
+func (n Notifier) notificationDriver() (extpoints.Driver, error) {
+	via, ok := n.configuration[Notify_Via]
+	if !ok {
+		return nil, errors.New("No notifier set")
+	}
+	driver := extpoints.Drivers.Lookup(via)
```

Comment:
use switch and simple methods call to notify lib instead of all these ext points and boiler plate.